scico/linop/xray/_xray.py

```diff
@@ -146,8 +151,11 @@ def _project(
         # ignored, while inds < 0 wrap around. So we set inds < 0 to ny.
         inds = jnp.where(inds >= 0, inds, ny)
 
+        # avoid incompatible types in the .add (scatter operation)
+        weights = weights.astype(im.dtype)
+
         y = (
-            jnp.zeros((len(angles), ny))
+            jnp.zeros((len(angles), ny), dtype=im.dtype)
```

Comment:
With respect to the longer comment below, this line (and others like it) seem to imply the intent to support different choices of `input_dtype` (and corresponding `output_dtype`).